llms/googleai/googleai_llm.go

```diff
@@ -0,0 +1,140 @@
+package googleai
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/google/generative-ai-go/genai"
```

Comment:
Note that this is the SDK for ai.google.dev, which doesn't use GCP auth but rather just an API key like OpenAI - you can get an API key from that website